astropy/wcs/src/wcslib_prjprm_wrap.c

```diff
@@ -0,0 +1,1025 @@
+        }
+    case 0:
+        result = Py_BuildValue("(OO)", prj_x1, prj_x2);
+        break;
+    default:
+        wcslib_prj_to_python_exc(status);
+        break;
+    }
+
+    exit:
+        Py_XDECREF(x1);
+        Py_XDECREF(x2);
+        Py_XDECREF(prj_x1);
+        Py_XDECREF(prj_x2);
+        Py_XDECREF(stat);
+
+    return result;
+}
+
+
+static PyObject* PyPrjprm_prjx2s(PyPrjprm* self, PyObject* args, PyObject* kwds)
+{
+    PyObject* x = NULL;
+    PyObject* y = NULL;
+    const char* keywords[] = { "x", "y", NULL };
+
+    if (is_prj_null(self)) return NULL;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "OO:prjx2s",
+        (char **)keywords, &x, &y)) {
+        return NULL;
+    }
+
+    if (self->x->prjx2s == NULL || self->x->flag == 0) {
+        if (is_readonly(self)) {
+            PyErr_SetString(
+                PyExc_AttributeError,
+                "Attribute 'prj' of 'astropy.wcs.Wcsprm.cel' objects is "
+                "read-only and cannot be automatically set.");
+            return NULL;
+        } else if (PyPrjprm_cset(self)) {
+            return NULL;
+        }
+    }
+
+    return _prj_eval(self, self->x->prjx2s, x, y);
+}
+
+
+static PyObject* PyPrjprm_prjs2x(PyPrjprm* self, PyObject* args, PyObject* kwds)
+{
+    PyObject* phi = NULL;
+    PyObject* theta = NULL;
+    const char* keywords[] = { "phi", "theta", NULL };
+
+    if (is_prj_null(self)) return NULL;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "OO:prjs2x",
+        (char **)keywords, &phi, &theta)) {
+        return NULL;
+    }
+
+    if (self->x->prjs2x == NULL || self->x->flag == 0) {
+        if (is_readonly(self)) {
+            PyErr_SetString(
+                PyExc_AttributeError,
+                "Attribute 'prj' of 'astropy.wcs.Wcsprm.cel' objects is "
+                "read-only and cannot be automatically set.");
+            return NULL;
+        } else if (PyPrjprm_cset(self)) {
+            return NULL;
+        }
+    }
+
+    return _prj_eval(self, self->x->prjs2x, phi, theta);
+}
+
+
+/***************************************************************************
+ * Member getters/setters (properties)
+ */
+
+static PyObject* PyPrjprm_get_flag(PyPrjprm* self, void* closure)
+{
+    if (is_prj_null(self)) {
+        return NULL;
+    } else {
+        return get_int("flag", self->x->flag);
+    }
+}
+
+
+static PyObject* PyPrjprm_get_code(PyPrjprm* self, void* closure)
+{
+    if (is_prj_null(self)) {
+        return NULL;
+    } else {
+        return get_string("code", self->x->code);
+    }
+}
+
+
+static int PyPrjprm_set_code(PyPrjprm* self, PyObject* value, void* closure)
+{
+    char code[4];
+    int code_len;
+
+    if (is_prj_null(self) || is_readonly(self)) {
+        return -1;
+    } else if (value == Py_None) {
+        if (strcmp("   ", self->x->code)) {
+            strcpy(self->x->code, "   ");
+            self->x->flag = 0;
+            if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
```

Comment:
Is there a scenario where owner exists but owner->x is a different pointer from self->x?

Reply:
Yes, owner of `Prjprm` is a `Celprm`. `x` of a `Prjprm` is a pointer to a WCSLIB's `prjprm` structure while the `x` of the owner (`Celprm`) is a pointer  to `celprm`.These are always different.